full-service/src/db/wallet_db_error.rs

```diff
@@ -0,0 +1,112 @@
+use displaydoc::Display;
+
+#[derive(Display, Debug)]
+pub enum WalletDbError {
```

Comment:
I moved several error types out of the top-level error.rs and into new files that are closer to where the error is generated. I think this helps organize the code more explicitly around API / Application Logic / Database layers.